src/memray/_ipython/flamegraph.py

```diff
@@ -118,6 +133,15 @@ def memray_flamegraph(self, line: str, cell: str) -> None:
             # It already printed a message, just return control to IPython.
             return
 
+        if options.temporal and options.temporary_allocation_threshold >= 0:
+            raise UsageError(
+                "Can't create a temporal flame graph of temporary allocations"
+            )
+        elif options.temporal or options.temporary_allocation_threshold >= 0:
+            file_format = "ALL_ALLOCATIONS"
+        else:
+            file_format = "AGGREGATED_ALLOCATIONS"
```

Comment:
Ah, this will be a nice improvement 👍  (I think we were not doing it already)

Reply:
💯

Reply:
Now that I think about it we could do it in the pytest plugin as well 🤔

Reply:
> Now that I think about it we could do it in the pytest plugin as well 🤔

https://github.com/bloomberg/pytest-memray/pull/107